.github/workflows/unittests.yml

```diff
@@ -34,5 +36,21 @@ jobs:
     - name: Compiling Babel translations
       run: |
         pybabel compile -f -d translations
-    - name: Run validations
-      run: build-tools/github/validate
+    - name: Check for changed files in grammar related directories
+      shell: pwsh
+      id: check_file_changed
+      run: |
+        $diff = git diff --name-only ${{ github.event.pull_request.base.sha }} ${{ github.sha }}
```

Comment:
haha this is pretty unreadable so I hope we never have to change this :D

Reply:
Maybe I should have added a comment. We basically extract all filenames from the differences between our current pull request last commit and the last commit on main.